test/LaunchDarkly.EventSource.Tests/EventSourceEndToEndTest.cs

```diff
@@ -29,100 +29,78 @@ public void ReceiveEventStreamInChunks()
             // parsed correctly regardless of how the chunks line up with the events.
 
             var eventData = new List<string>();
-            var chunks = new List<string>();
             for (var i = 0; i < 200; i++)
             {
                 eventData.Add(string.Format("data{0}", i) + new string('x', i % 7));
             }
             var allBody = string.Concat(eventData.Select(data => "data:" + data + "\n\n"));
-            for (var pos = 0; ;)
+            var allEventsReceived = new EventWaitHandle(false, EventResetMode.ManualReset);
+
+            IEnumerable<string> DoChunks()
             {
-                int i = chunks.Count;
-                int chunkSize = i % 20 + 1;
-                if (pos + chunkSize >= allBody.Length)
+                var i = 0;
+                for (var pos = 0; ;)
                 {
-                    chunks.Add(allBody.Substring(pos));
-                    break;
+                    int chunkSize = i % 20 + 1;
+                    if (pos + chunkSize >= allBody.Length)
+                    {
+                        yield return allBody.Substring(pos);
+                        break;
+                    }
+                    yield return allBody.Substring(pos, chunkSize);
+                    pos += chunkSize;
+                    i++;
```

Comment:
The logic here for making the chunked hasn't really changed, it's just that I switched from using an array of strings to having a generator function that `yield return`s the strings one at a time - not a big difference, just seemed a little cleaner.